python/cuspatial/cuspatial/core/spatial/distance.py

```diff
@@ -555,6 +556,78 @@ def pairwise_linestring_polygon_distance(
     )
 
 
+def pairwise_polygon_distance(polygons1: GeoSeries, polygons2: GeoSeries):
+    """Compute distance between pairs of (multi)polygons and (multi)polygons
```

Comment:
I would use an explicit pair notation instead for clarity and brevity:

```suggestion
    """Compute distance between (multi)polygon-(multi)polygon pairs."
```

hyphenate the pair relationship, and be consistent across APIs. So for `pairwise_linestring_polygon_distance` you would use "distance between linestring-polygon pairs".